src/slip.rs

```diff
@@ -0,0 +1,200 @@
+use secp256k1::{PublicKey};
+
+/// A UTXO slip containing record of owernship of funds on the network
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct Slip {
+    /// Contains concrete data which is not relative to state of the chain
+    body: SlipBody,
+    /// Flag determining if the `Slip` is part of the longest chain
+    lc: bool,
+    /// `SlipSpentStatus` indicating the state of the slip in the network
+    pub spent_status: SlipSpentStatus,
+}
+
+/// An object that holds concrete data not subjective to state of chain
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct SlipBody {
+    /// An `Sectp256K::PublicKey` determining who owns the `Slip`
+    add: PublicKey,
+    /// A enum brodcast type determining how to be processed by consensus
+    typ: SlipBroadcastType,
+    /// Amount of Saito
+    amt: u64,
+    /// `Block` id
+    bid: u64,
+    /// `Transaction` id
+    tid: u64,
+    /// `Slip` id
+    sid: u64,
+    /// The `Block` hash the slip originated from
+    bsh: [u8; 32],
```

Comment:
All of these can be computed from the location of the slip in the block encoding. I'm not convinced they even will be needed in the slip metadata until I see mempool algos. I strongly believe we should not include them in the slip body, they will waste bytes on disk and on the wire and make things slower. They can also cause potential bugs because they are duplicated state.

Reply:
Great point. Will move outside of the slip body as we still want to keep it around for checks with `UTXOSet`

Reply:
Thanks